src/query/storages/common/session/src/temp_table.rs

```diff
@@ -120,6 +133,9 @@ impl TempTblMgr {
                 true
             }
         };
+        if self.id_to_table.len() == 1 {
+            self.empty_state_changed = true;
```

Comment:
Is this redundant?

Reply:
no,  set to true when changing from empty to non-empty or reverse,  set to false when `is_empty` is called.
so caller add/remove mgr to map only when it is rue, do not have to lookup in map each time.